src/allocation_engine/interval_tree.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2
+
+/// Policy for resource allocation.
+#[derive(Copy, Clone, Debug)]
+pub enum AllocPolicy {
+    /// Allocate from the first matched entry.
```

Comment:
nit: `from the first matched entry` sounds a bit off. Maybe what we want to say is "allocate the first matched"?